components/script/dom/subtlecrypto.rs

```diff
@@ -827,25 +828,41 @@ impl SubtleCrypto {
             _ => return Err(Error::NotSupported),
         };
 
-        Ok(CryptoKey::new(
-            &self.global(),
-            KeyType::Secret,
-            extractable,
-            KeyAlgorithm { name },
-            usages,
-            handle,
-        ))
+        let key_algorithm = AesKeyAlgorithm {
+            parent: KeyAlgorithm { name: name.clone() },
+            length: key_gen_params.length,
+        };
+        let cx = GlobalScope::get_cx();
+        let crypto_key = unsafe {
+            rooted!(in(*cx) let mut algorithm: Value);
+            rooted!(in(*cx) let mut algorithm_object = ptr::null_mut::<JSObject>());
+            key_algorithm.to_jsval(*cx, algorithm.handle_mut());
+            algorithm_object.set(algorithm.to_object());
+
+            CryptoKey::new(
+                &self.global(),
+                KeyType::Secret,
+                extractable,
+                name,
+                algorithm_object.handle(),
+                usages,
+                handle,
+            )
```

Comment:
Perhaps there is an easier way to do this? I have a `AesKeyAlgorithm` and want to turn it into a `HandleObject`, but end up creating an intermediary js value that i call `to_object` on. Not very nice.

Reply:
There's https://doc.servo.org/script/dom/bindings/codegen/Bindings/SubtleCryptoBinding/struct.AesKeyAlgorithm.html#method.to_jsobject, but you need to start by creating a new JSObject (JS_NewObject) that you can pass as an argument. We should probably make that less of a foot gun.

Reply:
I see, thank you!